src/systemctl/systemctl.c

```diff
@@ -2679,13 +2680,81 @@ static const char *method_to_verb(const char *method) {
        return "n/a";
 }
 
+typedef struct {
+    sd_bus_slot *match;
+    sd_event *event;
+    Set *unit_paths;
+} WaitContext;
+
+static void wait_context_free(WaitContext *c) {
+    c->match = sd_bus_slot_unref(c->match);
+    c->event = sd_event_unref(c->event);
+    c->unit_paths = set_free(c->unit_paths);
+}
+
+static int on_properties_changed(sd_bus_message *m, void *userdata, sd_bus_error *error) {
+        WaitContext *c = userdata;
+        const char *path;
+        int r;
+
+        path = sd_bus_message_get_path(m);
+        if (!set_contains(c->unit_paths, path))
+                return 0;
+
+        /* Check if ActiveState changed to inactive/failed */
+        /* (s interface, a{sv} changed_properties, as invalidated_properties) */
+        r = sd_bus_message_skip(m, "s");
+        if (r < 0)
+                return bus_log_parse_error(r);
+        r = sd_bus_message_enter_container(m, SD_BUS_TYPE_ARRAY, "{sv}");
+        if (r < 0)
+                return bus_log_parse_error(r);
+
+        while ((r = sd_bus_message_enter_container(m, SD_BUS_TYPE_DICT_ENTRY, "sv")) > 0) {
+                const char *s;
+                r = sd_bus_message_read(m, "s", &s);
+                if (r < 0)
+                        return bus_log_parse_error(r);
+                if (streq(s, "ActiveState")) {
```

Comment:
This means that invoking `--wait` on a `RemainAfterExit=yes` unit will not return until `stop` is invoked on it. I think the documentation needs to be more explicit that it relates with systemd's concept of active-ness, rather with a naive "the unit terminates".